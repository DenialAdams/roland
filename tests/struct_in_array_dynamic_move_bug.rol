struct Bar {
   buf: &u8,
   buf_len: usize,
}

proc main() {
   let data = [
      Bar {
         buf: "test".pointer,
         buf_len: "test".length,
      },
      Bar {
         buf: "\n".pointer,
         buf_len: "\n".length,
      },
   ];

   println("---ptr");
   println(int_to_string(data[0].buf transmute usize extend u64));
   println("---len");
   println(int_to_string(data[0].buf_len transmute usize extend u64));
   println("---ptr2");
   println(int_to_string(data[1].buf transmute usize extend u64));
   println("---len2");
   println(int_to_string(data[1].buf_len transmute usize extend u64));
}
