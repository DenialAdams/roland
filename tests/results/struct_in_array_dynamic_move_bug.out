---ptr
10
---len
4
---ptr2
0
---len2
1
